File: data/code_pairs/cpp/2389.cpp
#include <iostream>
#include <vector>
#include <cassert>

// Main C++ Code
void sort_split_nodes(int numfn, int numelv, std::vector<std::vector<int>>& nfault, const std::vector<int>& indxiel) {
    if (numfn == 0) return;

    for (int i = 0; i < numfn; ++i) {
        nfault[0][i] = indxiel[nfault[0][i] - 1]; // Adjusted for 0-based indexing
    }
}

// Unit Test for the main C++ Code
void test_sort_split_nodes() {
    int numfn = 5;
    int numelv = 10;

    std::vector<std::vector<int>> nfault(3, std::vector<int>(numfn));
    std::vector<int> indxiel(numelv);

    // Populate nfault and indxiel with example values
    for (int i = 0; i < numfn; ++i) {
        nfault[0][i] = i + 1; // Adjust for 1-based to 0-based indexing
    }
    for (int i = 0; i < numelv; ++i) {
        indxiel[i] = i + 1;
    }

    // Call the function
    sort_split_nodes(numfn, numelv, nfault, indxiel);

    // Verify the results
    for (int i = 0; i < numfn; ++i) {
        assert(nfault[0][i] == indxiel[i]); // Verification based on 0-based indexing
    }

    std::cout << "All tests passed." << std::endl;
}

int main() {
    test_sort_split_nodes();
    return 0;
}

File: data/code_pairs/fortran/2389.f90
! Main Fortran Code
subroutine sort_split_nodes(nfault, indxiel, numfn, numelv)
    implicit none
    integer, intent(in) :: numfn, numelv
    integer, intent(inout) :: nfault(3, numfn), indxiel(numelv)
    integer :: i

    if (numfn .eq. 0) return
    do i = 1, numfn
        nfault(1, i) = indxiel(nfault(1, i))
    end do

end subroutine sort_split_nodes

! Unit Test for the main Fortran Code
program test_sort_split_nodes
    implicit none

    integer, parameter :: numfn = 5
    integer, parameter :: numelv = 10
    integer :: nfault(3, numfn)
    integer :: indxiel(numelv)
    integer :: i

    ! Initialize nfault and indxiel with test values
    do i = 1, numfn
        nfault(1, i) = i
    end do
    do i = 1, numelv
        indxiel(i) = i + 1
    end do

    ! Call the subroutine
    call sort_split_nodes(nfault, indxiel, numfn, numelv)

    ! Verify the results
    do i = 1, numfn
        if (nfault(1, i) /= indxiel(i)) then
            print *, 'Test failed at index', i
            stop
        end if
    end do
    print *, 'All tests passed.'
end program test_sort_split_nodes